src/settings.rs

```diff
@@ -5,6 +5,13 @@ use segment::common::cpu::get_num_cpus;
 use serde::Deserialize;
 use storage::types::StorageConfig;
 
+#[derive(Debug, Deserialize, Clone)]
+pub struct TlsConfig {
+    pub cert: String,
+    pub key: String,
+    pub client_cert: String,
+}
```

Comment:
Should we add `#[serde(default = ...)]` attributes or make these fields optional? 🤔

Reply:
I think it should be enough to make the whole config optional